packages/dao/contracts/Api3Template.sol

```diff
@@ -32,76 +38,120 @@ contract Api3Template is Api3BaseTemplate {
+        _validateVotingSettings(_mainVotingSettings);
+        _validateVotingSettings(_secondaryVotingSettings);
 
         (Kernel dao, ACL acl) = _createDAO();
-        (Api3Voting voting, Agent agent) = _setupApps(
-            dao, acl, _api3Pool, _votingSettings, _permissionManager
+        (Api3Voting mainVoting, Api3Voting secondaryVoting, Agent mainAgent, Agent secondaryAgent) = _setupApps(
+            dao, acl, _api3Pool, _mainVotingSettings,_secondaryVotingSettings
         );
-        _transferRootPermissionsFromTemplateAndFinalizeDAO(dao, _permissionManager);
+        _transferRootPermissionsFromTemplateAndFinalizeDAO(dao, mainVoting);
         _registerID(_id, dao);
 
         emit Api3DaoDeployed(
             address(dao),
             address(acl),
             address(_api3Pool),
-            address(voting),
-            address(agent)
+            address(mainVoting),
+            address(secondaryVoting),
+            address(mainAgent),
+            address(secondaryAgent)
         );
     }
 
     function _setupApps(
         Kernel _dao,
         ACL _acl,
         MiniMeToken _api3Pool,
-        uint64[3] memory _votingSettings,
-        address _permissionManager
+        uint64[3] memory _mainVotingSettings,
+        uint64[3] memory _secondaryVotingSettings
     )
     internal
-    returns (Api3Voting, Agent)
+    returns (Api3Voting, Api3Voting, Agent, Agent)
     {
-        Agent agent = _installDefaultAgentApp(_dao);
-        Api3Voting voting = _installVotingApp(_dao, _api3Pool, _votingSettings);
+        Agent mainAgent = _installDefaultAgentApp(_dao);
+        Agent secondaryAgent = _installNonDefaultAgentApp(_dao);
+        Api3Voting mainVoting = _installApi3VotingApp(_dao, _api3Pool, _mainVotingSettings);
+        Api3Voting secondaryVoting = _installApi3VotingApp(_dao, _api3Pool, _secondaryVotingSettings);
 
         _setupPermissions(
             _acl,
-            agent,
-            voting,
-            _permissionManager
+            mainAgent,
+            mainVoting,
+            secondaryAgent,
+            secondaryVoting,
+            mainVoting
         );
 
-        return (voting, agent);
+        return (mainVoting, secondaryVoting, mainAgent, secondaryAgent);
     }
 
     function _setupPermissions(
         ACL _acl,
-        Agent _agent,
-        Api3Voting _voting,
+        Agent _mainAgent,
+        Api3Voting _mainVoting,
+        Agent _secondaryAgent,
+        Api3Voting _secondaryVoting,
         address _permissionManager
     )
     internal
     {
-        _createAgentPermissions(_acl, _agent, _voting, _permissionManager);
-        _createVaultPermissions(_acl, Vault(_agent), _voting, _permissionManager);
+        _createAgentPermissions(_acl, _mainAgent, _mainVoting, _permissionManager);
+        _createAgentPermissions(_acl, _secondaryAgent, _secondaryVoting, _permissionManager);
+        _createVaultPermissions(_acl, Vault(_mainAgent), _mainVoting, _permissionManager);
```

Comment:
What is the implication on using the main pair here? For reference, we're planning to use the secondary agent for day-to-day transfers. Can we create vault permissions for both?

Reply:
Yep we can, added second permission, as I understand right should be granted to the `secondaryVotingApp` which is done as well...